package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,447 @@
+            return self.go_to_ts(ts)
+
+        self.reset_ts() # clear previous ts data
+        while self.step_to_frame(self.step+1, ts) == ts.frame:
+            self._read_next_step()
+            self.add_step_to_ts(ts.time)
+        self.ts_rep = self.calc_representative()
+        ts.aux.__dict__[self.name] = self.ts_rep
+        return ts
+
+
+    def step_to_frame(self, step, ts):
+        """ Calculate closest trajectory frame for auxiliary step *step*.
+
+        Calculated given dt and offset from *ts* as::
+            math.floor((self.times[step]-offset+ts.dt/2.)/ts.dt)
+        """
+        if step >= self.n_steps or step < 0:
+            ## make sure step is in the valid range. Raise error?
+            return None 
+        offset = ts.data.get('time_offset', 0)
+        return math.floor((self.times[step]-offset+ts.dt/2.)/ts.dt)
+
+    def go_to_ts(self, ts):
+        """ Move to and read auxiliary steps corresponding to *ts* """
+        # Need to define in each auxiliary reader
+        raise NotImplementedError(
+            "BUG: Override go_to_ts() in auxiliary reader!")
+
+    def reset_ts(self):
+        self.ts_data = np.array([])
+        self.ts_diffs = []
+
+    def add_step_to_ts(self, ts_time):
+        """ Add data from the current step to *ts_data* """
+        if len(self.ts_data) == 0:
+            self.ts_data = np.array([self.step_data])
+        else:
+            self.ts_data = np.append(self.ts_data, [self.step_data], axis=0)
+        self.ts_diffs.append(abs(self.time - ts_time))
+
+    def calc_representative(self):
+        """ Calculate a represenatative auxiliary value(s).
+        
+        Currently available options for calculating represenatative value are
+          *'closest': default; the value(s) from the step closest to in time to 
+           the trajectory timestep
+          *'average': average of the value(s) from steps 'assigned' to the 
+           trajectory timestep. 
+        Additionally, if a `cutoff` is specified, only steps within this time 
+        of the trajectory timestep are considered in calculating the 
+        represenatative.
+
+        Returns
+        -------
+        List (of length n_cols) of auxiliary value(s) 'representing' the 
+        timestep.
+        """
+
+        if self.cutoff != -1:
+            cutoff_data = np.array([self.ts_data[i] 
+                                    for i,d in enumerate(self.ts_diffs)
+                                    if d < self.cutoff])
+            cutoff_diffs = [d for d in self.ts_diffs if d < self.cutoff]
+        else:
+            cutoff_data = self.ts_data
+            cutoff_diffs = self.ts_diffs
+        if len(cutoff_data) == 0:
+            value = np.array([np.nan]*len(self.data_cols))
+        elif self.represent_ts_as == 'closest':
+            value = cutoff_data[np.argmin(cutoff_diffs),:]
+        elif self.represent_ts_as == 'average':
+            value = np.mean(cutoff_data, axis=0)
```

Comment:
Latter on `represent_ts_as` could take a callback or a method name as value. Also, you could have the methods in a dict like:

```python
_represent_ts_methods = {
    'closest': represent_as_closest,
    'average': represent_as_average,
}
```

With the method in functions like:

```python
def represent_as_closest(cutoff_data, cutoff_diffs):
    return cutoff_data[np.argmin(cutoff_diffs),:]

def represent_as_average(cutoff_data, cutoff_diffs):
    return np.mean(cutoff_data, axis=0)
```

This would allow to replace part of the code above by something roughly like:

```python
try:
    self._represent_ts_as_callable = self._represent_ts_methods[self.represent_ts_as]
except IndexError:
    if callable(self.represent_ts_as):
        self._represent_ts_as_callable = self.represent_ts_as
    else:
        raise ValueError('Invalid value for represent_ts_as. Must be a callable or one of {}'
            .format(' '.join(self._represent_ts_methods.keys()))
```

followed by:

```python
if len(cutoff_data) == 0:
    value = np.array([np.nan]*len(self.data_cols))
else:
    value = self._represent_ts_as_callable(cutoff_data, cutoff_diffs)
```

This would probably be a more flexible way to handle the representation. But it is not the priority.